core/server/common/src/main/java/alluxio/master/journal/raft/transport/CopycatGrpcClient.java

```diff
@@ -86,16 +97,28 @@ public CopycatGrpcClient(AlluxioConfiguration conf, UserState userState) {
       } catch (Throwable e) {
         throw new RuntimeException(e);
       }
+    }, mExecutor);
+    // When connection is build, complete the connection future with it on a catalyst thread context
+    // for setting up the connection.
+    buildFuture.whenComplete((result, error) -> {
+      threadContext.execute(() -> {
```

Comment:
What should we use the thread context executor for vs our own executor?

Reply:
End goal is to never use context executor within the transport layer for submitting work. Transport executor is for transport specific channel/connection building and termination.